contracts/simple-ica-controller/src/ibc.rs

```diff
@@ -127,19 +130,16 @@ fn acknowledge_dispatch(
     _caller: String,
     sender: String,
     callback_id: Option<String>,
-    msg: IbcPacketAckMsg,
+    ack: IbcPacketAckMsg,
 ) -> Result<IbcBasicResponse, ContractError> {
     let res = IbcBasicResponse::new().add_attribute("action", "acknowledge_dispatch");
     match callback_id {
         Some(id) => {
+            let msg: StdAck = from_slice(&ack.acknowledgement.data)?;
             // Send IBC packet ack message to another contract
             let res = res
                 .add_attribute("callback_id", &id)
-                .add_message(WasmMsg::Execute {
-                    contract_addr: sender,
-                    msg: to_binary(&ReceiveIbcResponseMsg { id, msg })?,
-                    funds: vec![],
-                });
+                .add_message(ReceiveIbcResponseMsg { id, msg }.into_cosmos_msg(sender)?);
```

Comment:
Should this be a sub-message with gas limit? Because simple-ica-controller has no control over what the callback capturer executes. I guess we don't want the relayer to be responsible for paying all the gas of something heavy.

Reply:
That is a very good design question. So far this is a "sample", but it would be good to raise such an issue as to be useful in production, we would need some limit

Reply:
For me the motivation would be to tell the reader: _Hey, look, this is something you need to think about!_

Reply:
Can add a comment. Something like:
```suggestion
                // In production, you may want to think about gas limits for this callback.
                .add_message(ReceiveIbcResponseMsg { id, msg }.into_cosmos_msg(sender)?);
```

Reply:
Leaving this unresolved, as this is something important to revisit.
Making an issue for it